### ISSUE TITLE
PFS_CACHE_SIZE should be adjustable

### ISSUE THREAD

**Issue by msteffen** [pachyderm/pachyderm#3323]
User is running into an issue where pachyderm pipelines consume a seemingly unbounded amount of memory and eventually OOM. I suspect that their hashtrees are growing over time, and causing the hashtree cache to consume an unbounded amount of memory. A way to test that would be to decrease PFS_CACHE_SIZE and see if that reduces the severity of the memory issue, but there's no way to set this value in e.g. the pipeline spec (you can `kubectl edit` the pipeline rc, which we did).

**Comment by ysimonson** [comment]
Is it not adjustable as an environment variable? Or is it that it should be adjustable via alternate means as well, such as in the pipeline spec?

**Comment by jmcgill** [comment]
I do not believe (in 1.7.5) that it is possible to configure this *or* a hard memory limit on the storage pods via the pipeline spec. 

We can manually set these values via kubectl on each pod, or on the replication controller to affect all newly created pods, however every deploy will wipe these values out.
